src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleWrapper.cs

```diff
@@ -177,75 +235,285 @@ public void CancelDiscovery()
+            ITestHostLauncher customTestHostLauncher)
         {
             var testCaseList = testCases.ToList();
-            this.testPlatformEventSource.TranslationLayerExecutionStart(1, 0, testCaseList.Count, runSettings ?? string.Empty);
+            this.testPlatformEventSource.TranslationLayerExecutionStart(
+                1,
+                0,
+                testCaseList.Count,
+                runSettings ?? string.Empty);
 
             this.EnsureInitialized();
-            this.requestSender.StartTestRunWithCustomHost(testCaseList, runSettings, options: null, runEventsHandler: testRunEventsHandler, customTestHostLauncher: customTestHostLauncher);
+            this.requestSender.StartTestRunWithCustomHost(
+                testCaseList,
+                runSettings,
+                options,
+                testSessionInfo,
+                testRunEventsHandler,
+                customTestHostLauncher);
         }
 
         /// <inheritdoc/>
-        public void RunTestsWithCustomTestHost(IEnumerable<TestCase> testCases, string runSettings, TestPlatformOptions options, ITestRunEventsHandler testRunEventsHandler, ITestHostLauncher customTestHostLauncher)
+        public ITestSession StartTestSession(
+            IList<string> sources,
+            string runSettings,
+            ITestSessionEventsHandler eventsHandler)
         {
-            var testCaseList = testCases.ToList();
-            this.testPlatformEventSource.TranslationLayerExecutionStart(1, 0, testCaseList.Count, runSettings ?? string.Empty);
+            return this.StartTestSession(
+                sources,
+                runSettings,
+                options: null,
+                eventsHandler);
+        }
+
+        /// <inheritdoc/>
+        public ITestSession StartTestSession(
+            IList<string> sources,
+            string runSettings,
+            TestPlatformOptions options,
+            ITestSessionEventsHandler eventsHandler)
+        {
+            return this.StartTestSession(
+                sources,
+                runSettings,
+                options,
+                eventsHandler,
+                testHostLauncher: null);
+        }
+
+        /// <inheritdoc/>
+        public ITestSession StartTestSession(
+            IList<string> sources,
+            string runSettings,
+            TestPlatformOptions options,
+            ITestSessionEventsHandler eventsHandler,
+            ITestHostLauncher testHostLauncher)
+        {
+            this.testPlatformEventSource.TranslationLayerStartTestSessionStart();
+
+            this.EnsureInitialized();
+            return new TestSession(
+                this.requestSender.StartTestSession(
+                    sources,
+                    runSettings,
+                    options,
+                    eventsHandler,
+                    testHostLauncher),
+                this);
+        }
+
+        /// <inheritdoc/>
+        public bool StopTestSession(
```

Comment:
It's necessary to have it exposed on `IVsTestConsoleWrapper` because `ITestSession` is just a wrapper over it and the calls get forwarded to the underlying `IVsTestConsoleWrapper` object. In this way we also keep the symmetry of the API (we have both `StopTestSession` and `StartTestSession` on the same `IVsTestConsoleWrapper` object).

I decided to expose it on `ITestSession` for the convenience of the caller as well as because you'll need to actually pass a `TestSessionInfo` object to the underlying `IVsTestConsoleWrapper`, however this object is internally managed by the `ITestSession` and it's private.